hikari/impl/entity_factory.py

```diff
@@ -1451,6 +1451,70 @@ def deserialize_invite_with_metadata(self, payload: data_binding.JSONObject) ->
+            type=message_models.MessageActivityType(payload["type"]), party_id=payload.get("party_id")
+        )
+
+    def _deserialize_message_application(self, payload: data_binding.JSONObject) -> message_models.MessageApplication:
+        primary_sku_id: typing.Optional[snowflakes.Snowflake] = None
+        if (raw_primary_sku_id := payload.get("primary_sku_id")) is not None:
+            primary_sku_id = snowflakes.Snowflake(raw_primary_sku_id)
+
+        return message_models.MessageApplication(
+            id=snowflakes.Snowflake(payload["id"]),
+            name=payload["name"],
+            description=payload["description"],
+            icon_hash=payload["icon"],
+            summary=payload["summary"],
+            cover_image_hash=payload["cover_image"],
+            primary_sku_id=primary_sku_id,
+        )
+
+    def _deserialize_message_attachment(self, payload: data_binding.JSONObject) -> message_models.Attachment:
+        return message_models.Attachment(
+            id=snowflakes.Snowflake(payload["id"]),
+            filename=payload["filename"],
+            size=int(payload["size"]),
+            url=payload["url"],
+            proxy_url=payload["proxy_url"],
+            height=payload.get("height"),
+            width=payload.get("width"),
+        )
+
+    def _deserialize_message_reaction(self, payload: data_binding.JSONObject) -> message_models.Reaction:
+        return message_models.Reaction(
+            count=int(payload["count"]), emoji=self.deserialize_emoji(payload["emoji"]), is_me=payload["me"]
+        )
+
+    def _deserialize_message_reference(self, payload: data_binding.JSONObject) -> message_models.MessageReference:
+        message_reference_message_id: typing.Optional[snowflakes.Snowflake] = None
+        if "message_id" in payload:
+            message_reference_message_id = snowflakes.Snowflake(payload["message_id"])
+
+        message_reference_guild_id: typing.Optional[snowflakes.Snowflake] = None
+        if "guild_id" in payload:
+            message_reference_guild_id = snowflakes.Snowflake(payload["guild_id"])
+
+        return message_models.MessageReference(
+            app=self._app,
+            id=message_reference_message_id,
+            channel_id=snowflakes.Snowflake(payload["channel_id"]),
+            guild_id=message_reference_guild_id,
+        )
+
+    def _deserialize_sticker(self, payload: data_binding.JSONObject) -> message_models.Sticker:
+        return message_models.Sticker(
+            id=snowflakes.Snowflake(payload["id"]),
+            pack_id=snowflakes.Snowflake(payload["pack_id"]),
+            name=payload["name"],
+            description=payload["description"],
+            tags=[tag.strip() for tag in payload["tags"].split(",")] if "tags" in payload else [],
+            asset_hash=payload["asset"],
+            preview_asset_hash=payload["preview_asset"],
+            format_type=message_models.StickerFormatType(payload["format_type"]),
+        )
+
```

Comment:
These should be public for consistency sake

Reply:
The approach this marshaller was built on was to only make deserthey ialize methods public if are specifically called from somewhere rather than only internally within the factory rather than keeping it 1-to-1 relationship for models to deserialize/serialize methods but this didn't stop the use of private methods for avoiding duplication.
And the only place stickers are documented is on message objects right now